lib/ash/changeset/changeset.ex

```diff
@@ -1094,50 +1094,58 @@ defmodule Ash.Changeset do
   end
 
   @doc false
+  # Returns either an appropriate expression for an atomic condition or a value
+  # indicated that the condition cannot be handled atomically.
+  #
+  # Validation logic matches on failure. So, for example, `present(:field)` is
+  # going to _match_ when `:field` is `nil`. However, when applying this logic
+  # to a `where` condition, the opposite is desired. The end result is kinda
+  # ugly because it can end up reading like "not is not equal to" but
+  # ultimately produces the correct results.
+  @spec atomic_condition([{module(), keyword()}], Ash.Changeset.t(), map()) ::
+          {:atomic, Ash.Expr.t() | boolean()} | {:not_atomic, String.t()}
   def atomic_condition(where, changeset, context) do
     Enum.reduce_while(where, {:atomic, true}, fn {module, validation_opts},
-                                                 {:atomic, condition} ->
+                                                 {:atomic, condition_expr} ->
```

Comment:
I renamed this variable because `condition` coupled with it starting as `true` was _really_ confusing at first glance. I get it now, but I think this slight change improves the clarity.